deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,328 @@
+# ----------------------------------------------------------------------------
+#
+"""Module of weak segments performance check."""
+from collections import defaultdict
+from typing import Callable, Dict, List, Union
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import sklearn
+from category_encoders import TargetEncoder
+from packaging import version
+from sklearn.model_selection import GridSearchCV
+from sklearn.tree import DecisionTreeRegressor
+
+from deepchecks import ConditionCategory, ConditionResult, Dataset
+from deepchecks.core import CheckResult
+from deepchecks.core.check_result import DisplayMap
+from deepchecks.core.errors import DeepchecksNotSupportedError, DeepchecksProcessError
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.tabular.context import _DummyModel
+from deepchecks.tabular.utils.task_type import TaskType
+from deepchecks.utils.performance.partition import (convert_tree_leaves_into_filters,
+                                                    partition_numeric_feature_around_segment)
+from deepchecks.utils.single_sample_metrics import calculate_per_sample_loss
+from deepchecks.utils.strings import format_number, format_percent
+from deepchecks.utils.typing import Hashable
+
+__all__ = ['WeakSegmentsPerformance']
+
+
+class WeakSegmentsPerformance(SingleDatasetCheck):
+    """Search for segments with the low performance score based on defined scorer function.
+
+    Parameters
+    ----------
+    columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to check, if none are given checks all columns except ignored ones.
+    ignore_columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to ignore, if none given checks based on columns variable
+    n_top_features : int , default: 5
+        Number of features to use for segment search. Top columns are selected based on feature importance.
+    segment_minimum_size_ratio: float , default: 0.05
+        Minimum size ratio for segments. Will only search for segments of
+        size >= segment_minimum_size_ratio * data_size.
+    alternative_scorer : Tuple[str, Union[str, Callable]] , default: None
+        Score to show, either function or sklearn scorer name.
+        If None, a default scorer (per the model type) will be used.
+    loss_per_sample: Union[np.array, pd.Series, None], default: None
+        Loss per sample used to detect relevant weak segments. If None, the check calculates loss per sample by using
+         for classification and mean square error for regression.
+    n_samples : int , default: 10_000
+        number of samples to use for this check.
+    n_to_show : int , default: 3
+        number of segments with the weakest performance to show.
+    categorical_aggregation_threshold : float , default: 0.05
+        Categories with appearance ratio below threshold will be merged into "Other" category.
```

Comment:
"appearance ratio"? I don't think that a good, intuitive phrase for that. 
I guess you didn't want to use "max_num_categories" which is easier to explain but less methodologically justified. 
Still, a different explanation is needed here.